onnxruntime/python/tools/transformers/models/llama/README.md

```diff
@@ -17,12 +17,31 @@ $ python3 -m onnxruntime.transformers.models.llama.convert_to_onnx -m meta-llama
 
 To make this option compatible with [Hugging Face's Optimum](https://github.com/huggingface/optimum), you will need to create `config.json` and `generation_config.json` for your model and store them in the same directory as your ONNX models. For example, you can find those JSON files for LLaMA-2 7B on Hugging Face [here](https://huggingface.co/meta-llama/Llama-2-7b-hf).
 
+As indicated in `requirements.txt`, you will also need to install Optimum from source. Once installed, you will need to modify `ORTModelForCausalLM.forward` in `optimum/optimum/onnxruntime/modeling_decoder.py` as follows:
+
+```
+# Before
+if self.use_cache:
+    if past_key_values is not None:
+        input_ids = input_ids[:, -1:]
+        # Flatten the past_key_values (no need to flatten for models using multi-query attn)
+
+
+# After
+if self.use_cache:
+    if past_key_values is not None:
+        input_ids = input_ids[:, -1:] if past_key_values[0][0].shape[2] != 0 else input_ids
+        # Flatten the past_key_values (no need to flatten for models using multi-query attn)
+```
+
 ### Option 2: from [Microsoft's custom export](https://github.com/microsoft/Llama-2-Onnx)
 
 Please follow the [README instructions](https://github.com/microsoft/Llama-2-Onnx#before-you-start) in the custom export of LLaMA-2.
 
 ### Option 3: from [Hugging Face Optimum](https://github.com/huggingface/optimum)
 
+Note that this will produce two ONNX models whereas the above two options produce one ONNX model. 
```

Comment:
With latest optimum installation which includes the changes from [https://github.com/huggingface/optimum/pull/1257](https://github.com/huggingface/optimum/pull/1257). 

The following code snippet just conduct one model which contains kv_cache.

```python
model = ORTModelForCausalLM.from_pretrained(
    name,
    export=True,
    use_auth_token=True,
)
```

Reply:
`legacy=True` should be work to conduct the models with and w/o kv cache in once run.

Reply:
This documentation was added before those changes were merged into Optimum. Since previous Optimum versions may export two ONNX models by default, the README can be updated to reflect that. As mentioned [here](https://github.com/microsoft/onnxruntime/pull/18021#discussion_r1365089506), I will update the README once another Optimum PR is merged.